grpc-proto/pom.xml

```diff
@@ -54,6 +68,15 @@
           <protocArtifact>com.google.protobuf:protoc:3.12.0:exe:${os.detected.classifier}</protocArtifact>
           <pluginId>grpc-java</pluginId>
           <pluginArtifact>io.grpc:protoc-gen-grpc-java:1.35.0:exe:${os.detected.classifier}</pluginArtifact>
+          <protocPlugins>
+            <protocPlugin>
+              <id>reactor-grpc</id>
```

Comment:
we might consider creating a dedicated module `grpc-proto-reactive` for reactive gRPC support to not require our clients to add import reactive libraries (via transitive dependencies) if they do not need reactive support.

Reply:
this seems as a good idea..

Reply:
yeah, but it is not as simple as it sounds so I think we may consider doing it in a separate task.
I've tried to create a new module `grpc-proto-reactive`, but this module need to have both `proto` files (`stargate.proto` and `query.proto`). 
I've tried to make the new module inherit the `grpc-proto`:
```
    <dependencies>
        <dependency>
            <groupId>io.projectreactor</groupId>
            <artifactId>reactor-core</artifactId>
            <version>3.4.13</version>
        </dependency>
    </dependencies>

    <build>
        <extensions>
            <extension>
                <groupId>kr.motd.maven</groupId>
                <artifactId>os-maven-plugin</artifactId>
                <version>1.6.2</version>
            </extension>
        </extensions>
        <plugins>
            <plugin>
                <groupId>org.xolstice.maven.plugins</groupId>
                <artifactId>protobuf-maven-plugin</artifactId>
                <version>0.6.1</version>
                <configuration>
                    <protoSourceRoot>proto</protoSourceRoot>
                    <protocArtifact>com.google.protobuf:protoc:3.12.0:exe:${os.detected.classifier}</protocArtifact>
                    <pluginId>grpc-java</pluginId>
                    <pluginArtifact>io.grpc:protoc-gen-grpc-java:1.35.0:exe:${os.detected.classifier}</pluginArtifact>
                    <protocPlugins>
                        <protocPlugin>
                            <id>reactor-grpc</id>
                            <groupId>com.salesforce.servicelibs</groupId>
                            <artifactId>reactor-grpc</artifactId>
                            <version>${grpc.reactor.version}</version>
                            <mainClass>com.salesforce.reactorgrpc.ReactorGrpcGenerator</mainClass>
                        </protocPlugin>
                    </protocPlugins>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>compile</goal>
                            <goal>compile-custom</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>
```
So it would have structure like `grpc-proto-reactive` -> `grpc-proto` -> `stargate parent` but it does not seem to work:
```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[WARNING] 'parent.relativePath' of POM io.stargate.grpc:grpc-proto-reactive:1.0.45-SNAPSHOT (/Users/tomasz.lelek/IntelliJ_workspace/stargate/grpc-proto-reactive/pom.xml) points at io.stargate:stargate instead of io.stargate.grpc:grpc-proto, please verify your project structure @ line 6, column 13
[FATAL] Non-resolvable parent POM for io.stargate.grpc:grpc-proto-reactive:1.0.45-SNAPSHOT: Could not find artifact io.stargate.grpc:grpc-proto:pom:1.0.45-SNAPSHOT in datastax-artifactory (https://repo.sjc.dsinternal.org/artifactory/dse) and 'parent.relativePath' points at wrong local POM @ line 6, column 13
```
Adding the:
```
  <dependency>
            <groupId>io.stargate.grpc</groupId>
            <artifactId>grpc-proto</artifactId>
            <version>${project.version}</version>
        </dependency>
```
as a dependency also does not work because `proto` files are not imported. We would need to figure this out.